### PR TITLE
Hex symbol sizes and Windows path incompatibility

### PR DESCRIPTION
**Pull request by Panometric**
I think others may benefit from these.  The program was not useful to me without them.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
retrigger CI

File: WCS.py [hunk 1]
```diff
@@ -148,17 +148,21 @@ def read_su(self, tu: str) -> None:
         :param tu: the translation unit
         :return:
         """
+    # Needs to be able to handle both cases, i.e.: 
+    #   c:\\userlibs\\gcc\\arm-none-eabi\\include\\assert.h:41:6:__assert_func	16	static
+    #   main.c:113:6:vAssertCalled	8	static
+    # Now Matches seven groups https://regex101.com/r/DsvQv6/7
 
-        su_line = re.compile(r'^([^ :]+):([\d]+):([\d]+):(.+)\t(\d+)\t(\S+)$')
+        su_line = re.compile(r'^(([^\\]+\\)*[^:]*):([\d]+):([\d]+):(.+)\t(\d+)\t(\S+)$')
```
Comment (ericLemanissier) [inline WCS.py]:
```suggestion
        su_line = re.compile(r'^(.+):(\d+):(\d+):(.+)\t(\d+)\t(\S+)$')
```

File: WCS.py [hunk 2]
```diff
@@ -148,17 +148,21 @@ def read_su(self, tu: str) -> None:
...
         i = 1
 
         with open(tu[0:tu.rindex(".")] + su_ext, "rt", encoding="latin_1") as file_:
             for line in file_:
                 m = su_line.match(line)
                 if m:
-                    fxn = m.group(4)
+                    fxn = m.group(5)
```
Comment (ericLemanissier) [inline WCS.py]:
```suggestion
                    fxn = m.group(4)
```

**Review by ericLemanissier — COMMENTED** [review]
what do you think of this regex simplification ?

File: WCS.py [hunk 3]
```diff
@@ -148,17 +148,21 @@ def read_su(self, tu: str) -> None:
         :param tu: the translation unit
         :return:
         """
+    # Needs to be able to handle both cases, i.e.: 
+    #   c:\\userlibs\\gcc\\arm-none-eabi\\include\\assert.h:41:6:__assert_func	16	static
+    #   main.c:113:6:vAssertCalled	8	static
+    # Now Matches seven groups https://regex101.com/r/DsvQv6/7
```
Comment (ericLemanissier) [inline WCS.py]:
```suggestion
    # Now Matches six groups https://regex101.com/r/Imi0sq/1
```

File: WCS.py [hunk 4]
```diff
@@ -148,17 +148,21 @@ def read_su(self, tu: str) -> None:
...
 
         with open(tu[0:tu.rindex(".")] + su_ext, "rt", encoding="latin_1") as file_:
             for line in file_:
                 m = su_line.match(line)
                 if m:
-                    fxn = m.group(4)
+                    fxn = m.group(5)
                     fxn_dict2 = self.find_demangled_fxn(tu, fxn)
-                    fxn_dict2['local_stack'] = int(m.group(5))
+                    fxn_dict2['local_stack'] = int(m.group(6))
```
Comment (ericLemanissier) [inline WCS.py]:
```suggestion
                    fxn_dict2['local_stack'] = int(m.group(5))
```

File: WCS.py (hunk 1 above)
Comment (Panometric) [reply WCS.py]: Cleaner 👍

**Comment by ericLemanissier** [comment]
@Panometric  Can you try with the current PR, and tell us if it works for you ?

**Comment by Panometric** [comment]
@ericLemanissier sorry I did not get back to you sooner. Yes this works well. Thanks for improving it.

**Comment by ericLemanissier** [comment]
Thanks for your feedback !

**Pull request merged**